src/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/DiagnosticSourceEventSource.cs

```diff
@@ -484,9 +484,7 @@ public static void DestroyFilterAndTransformList(ref FilterAndTransform specList
             /// </summary>
             public FilterAndTransform(string filterAndPayloadSpec, int startIdx, int endIdx, DiagnosticSourceEventSource eventSource, FilterAndTransform next)
             {
-#if DEBUG
-                string spec = filterAndPayloadSpec.Substring(startIdx, endIdx - startIdx);
-#endif 
+                Debug.Assert(filterAndPayloadSpec != null && startIdx <= endIdx && endIdx <= filterAndPayloadSpec.Length);
```

Comment:
Should we also add `>= 0` for start/end index.

Reply:
Sure.